.github/workflows/build_and_test.yml

```diff
@@ -1,7 +1,5 @@
-name: Build and test [Python 3.9, 3.10]
-
+name: Build and test
 on: [push, pull_request]
```

Comment:
Re my comments a few weeks ago about unnecessary compute time, from a read of the [updated workflow docs](https://docs.github.com/en/actions), I think we should be able to condition tests to run only when relevant files are affected.

E.g., something like:
```
on:
  [push, pull_request]:
    paths:
      - './ogusa/**.py'
      -  './tests/**.py'
```

Reply:
@jdebacker. Yes. I'll add this. But we probably need to expand beyond the `.py` files.

- In the `./ogusa/` directory, we have all `.py` files except for one `.json` file. I think we need to add the `.json` file to the directory tracking.

- In the `./tests/` directory, we probably also need to test changes in the `.pkl` files and `.csv` files in the `./tests/test_io_data/` directory.

- In the main directory, we probably need to run the tests whenever `environment.yml`, `pyproject.toml`, `pytest.ini`, or `setup.py` changes.